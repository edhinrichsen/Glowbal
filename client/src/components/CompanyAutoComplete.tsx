import React, { useState } from 'react';

import { AutoComplete } from 'antd';


type CompanyAutoCompleteProps = {
    /** Callback function for when a company is selected */
    onSelect?: (value: string) => void;
    /** Callback function for when the input field is changed */
    onChange?: (value: string) => void;
    //** Initial value for the input field */
    initialValue?: string;
}

function CompanyAutoComplete(props: CompanyAutoCompleteProps) {
    
    const [options, setOptions] = useState<Array<any>>();
    const [value, setValue] = useState(props.initialValue ? props.initialValue : "");
    
    async function fetchCompanyAutoComplete(query: string): Promise<Array<any>> {
        const res = await fetch ('https://autocomplete.clearbit.com/v1/companies/suggest?query=' + query);
        return res.json();
    }
  
    async function searchResult(query: string) {

        const data = await fetchCompanyAutoComplete(query);
    
        return data
            .map((item: any, idx: any) => {
                return ({
                    value: (`${item.name}`),
                    label: (
                        <div style={{ padding: 5 }}>
                            <span style={{ paddingRight: 10 }}>
                                <img alt=" " src={item.logo} width="20px" style={{borderRadius: 10}}/>
                            </span>
                            <span>
                                {item.name}
                            </span>
                        </div>),
                });
        });
    };

    const handleValueChange = (value: string) => {
        setValue(value);
        if (props.onChange) {
            props.onChange(value);
        }
    }

    async function handleSearch(value: string) { 
      setOptions(value ? await searchResult(value) : []);
    };
  
    const onSelect = (value: string) => {
      handleValueChange(value);
      if (props.onSelect) {
        props.onSelect(value)
      }
    };

    return (
      <AutoComplete
        dropdownMatchSelectWidth={252}
        style={{ width: 300 }}
        options={options}
        onSelect={onSelect}
        onSearch={handleSearch}
        onChange={handleValueChange}
        value={value}
      />
      
    );
}

export default CompanyAutoComplete; 