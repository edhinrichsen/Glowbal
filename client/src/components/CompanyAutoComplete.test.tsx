import React from 'react';
import { shallow } from 'enzyme';
import CompanyAutoComplete from './CompanyAutoComplete';

describe('Testing CompanyAutoComplete', () => {

  it('renders without crashing', () => {
    const component = shallow(<CompanyAutoComplete />);
    expect(component).toMatchSnapshot();
    component.unmount();
  });
    
})
