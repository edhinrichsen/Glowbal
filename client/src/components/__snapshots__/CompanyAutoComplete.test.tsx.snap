// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CompanyAutoComplete renders without crashing 1`] = `
<ForwardRef(AutoComplete)
  dropdownMatchSelectWidth={252}
  onChange={[Function]}
  onSearch={[Function]}
  onSelect={[Function]}
  style={
    Object {
      "width": 300,
    }
  }
  value=""
/>
`;
